homeassistant/components/acmeda/helpers.py

```diff
@@ -0,0 +1,81 @@
+"""Helper functions for Acmeda Pulse."""
+from homeassistant import config_entries
+from homeassistant.helpers.device_registry import async_get_registry as get_dev_reg
+from homeassistant.helpers.entity_registry import async_get_registry as get_ent_reg
+
+from .const import DOMAIN, LOGGER
+
+
+async def update_entities(
+    hass, entity_class, config_entry, current, async_add_entities
+):
+    """Add any new entities, remove any old ones."""
+    if config_entry.entry_id not in hass.data[DOMAIN]:
+        LOGGER.error("Invalid config_entry %s", config_entry.entry_id)
+        return
+
+    hub = hass.data[DOMAIN][config_entry.entry_id]
+    LOGGER.debug("Looking for new %s on: %s", entity_class.__name__, hub.host)
+
+    api = hub.api.rollers
+
+    new_items = []
+    for unique_id, roller in api.items():
+        if unique_id not in current:
+            LOGGER.debug("New %s %s", entity_class.__name__, unique_id)
+            new_item = entity_class(hass, roller)
+            current[unique_id] = new_item
```

Comment:
Don't store the entity in a shared container. Entities are not allowed to be accessed outside their platforms and we should also not access an entity from outside itself directly so there shouldn't be a need to store them.

It's ok to store a reference to the entity eg the entity_id.

Reply:
Done, now only storing unique_id of the entity.